internal/backend/db/dbgorm/connections.go

```diff
@@ -82,6 +96,10 @@ func (db *gormdb) ListConnections(ctx context.Context, filter sdkservices.ListCo
 		q = q.Where("project_id = ?", filter.ProjectID.UUIDValue())
 	}
 
+	if filter.StatusCode != sdktypes.StatusCodeUnspecified {
```

Comment:
you need to consider indexing this field as well
as well as integration id

Reply:
fixed